bin/prover-client/src/operators/cl_agg.rs

```diff
@@ -29,28 +31,33 @@ impl ClAggOperator {
 
 impl ProvingOp for ClAggOperator {
     type Prover = ClAggProver;
-    type Params = (u64, u64);
+    type Params = Vec<(L2BlockId, L2BlockId)>;
 
     async fn create_task(
         &self,
-        params: (u64, u64),
+        batches: Self::Params,
         task_tracker: Arc<Mutex<TaskTracker>>,
         db: &ProofDb,
     ) -> Result<Vec<ProofKey>, ProvingTaskError> {
-        let (start_height, end_height) = params;
+        let mut cl_stf_deps = Vec::with_capacity(batches.len());
 
-        let len = (end_height - start_height) as usize + 1;
-        let mut cl_stf_deps = Vec::with_capacity(len);
+        // Extract first and last block IDs from batches, error if empty
+        let (start_blkid, end_blkid) = match (batches.first(), batches.last()) {
+            (Some(first), Some(last)) => (first.0, last.1),
```

Comment:
I'm unsure of if this is the best way to structure the proof params here.  Isn't this aggregation always "one epoch at a time", extending the proof we have for the previous epoch by the next epoch we just made?

Reply:
Yes, `cl_agg` is one epoch at a time. 
This `create_task` allows the creator to specify blocks in mini-batches that can be proven independently and subsequently aggregated.